### DISCUSSION TITLE
Exporting webrtc implementation out of it.

### DISCUSSION THREAD

**Discussion by ayazalavi** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1510] [Ideas]
I need webrtc implementation for esp boards for one of my projects.  

Can anyone let me know how to export webrtc out of this project into my own.? I have my own signalling server, turn and stun servers.

**Comment by hassanctech** [comment]
This is our low level SDK, any porting you would need to do on your own.  In many places OS specific functionality can be custom defined so this should be possible but you would need to do it in your own.  If you have a pointed question we can provide some guidance.

Regarding using your own signaling server and turn servers, this is not explicitly supported.

**Reply by ayazalavi** [reply]
Can you provide simple steps how to replace signaling server and also how to provide my own turn server? files to look at etc.

**Reply by hassanctech** [reply]
The SDK has build in state machines which go through our backend APIs, so replacing them all with your own is not something I can provide any guidance on.

What I meant was if you have specific questions porting to esp boards if you want to override functionality (like a threading library or something) I can point you to how to do that.

**Reply by ayazalavi** [reply]
I have exported it finally, thanks for the response Can you look at my next comment please? I am looking to connect this with adf library. It is audio framework provided by esp.

**Comment by ayazalavi** [comment]
Is there any way to use esp ADF with this repository? 

see examples repository https://github.com/espressif/esp-adf/tree/master/examples/player

which one is best reference for using in this project?

Edited:
Is there any way to get handle to the socket to read data from and write data to? How does this work in this library?
